test/functorch/test_control_flow.py

```diff
@@ -8563,6 +8567,249 @@ def false_fn(x, y):
         self.assertEqual(compiled_out, out)
 
 
+class TestAutoFunctionalizeControlFlow(TestCase):
+    def check(
+        self, gen_fn, args, device, dynamic
+    ) -> tuple[torch.fx.GraphModule, torch.fx.GraphModule]:
+        from unittest.mock import patch
+
+        args = pytree.tree_map(lambda t: t.to(device=device), args)
+
+        def _clone(args):
+            return [
+                arg.clone() if isinstance(arg, torch.Tensor) else arg for arg in args
+            ]
+
+        def _new_fn():
+            mod_or_fn = gen_fn()
+            if isinstance(mod_or_fn, torch.nn.Module):
+                mod_or_fn.to(device)
+            return mod_or_fn
+
+        with patch.object(
+            torch._dynamo.variables.higher_order_ops.CondHigherOrderVariable,
+            "supports_input_mutation",
+            True,
```

Comment:
`supports_input_mutation` is on the VariableTracker? Ideally it is on the HOP

Reply:
Yeah, that's a good point, I can do a refactoring.